### ISSUE TITLE
[DeviceClient] Cannot set trusted certificate path when using MQTT

### ISSUE THREAD

**Issue by timtay-microsoft** [Azure/azure-iot-sdk-java#342]
- **OS and version used:** Windows 10

- **Java runtime used:** 8

- **SDK version used:** 1.14.0

# Description of the issue:
The SetOption API allows a device client object to set the certificate path of a custom trusted certificate. However, the API throws an exception if the device client is using MQTT or MQTT_WS, even though it would work.

Also, the DeviceClient doesn't allow you to use the SetCertificateAuthority option like ModuleClient does. 

# Code sample exhibiting the issue:

```
DeviceClient client = new DeviceClient("connection string", MQTT)
client.setOption("SetCertificatePath", "some trusted cert's path"); //throws
client.setOption("SetCertificateAuthority", "some trusted cert"); //throws
```

**Comment by timtay-microsoft** [comment]
This is a bug on our side, and we will be getting the fix out soon.

**Comment by timtay-microsoft** [comment]
The fix for this issue has been checked in as of 3cacc197d1c0139d12eabb5c54bab7fd22ee151d

**Comment by timtay-microsoft** [comment]
The fix for this issue has been released as of device client 1.14.1
